### ISSUE TITLE
Cookie Consent Pops up on every page

### ISSUE THREAD

**Issue by Tigerlady** [MasaCMS/MasaCMS#297]
**Describe the bug**
I recently updated a site to the latest version of Masa that uses Cookie Consent, and it now pops up on every single page when browsing the site. I haven't had this happen before.   I am not sure this is a Masa issue, but it only started after updating to the latest version.

**To Reproduce**
Create a Cookie Consent as a component
Turn on Cookie Consent in the Site contentrender.cfc
Browse the site like normal. The Cookie Consent pops up on every page even if you accept the cookie consent. 
When looking at the cookies on the site, I see that it has been set with a value of yes.  "YES%2020240321-9-32"
With this value of yes, it shouldn't appear again. 

If you clear your cookies and refresh the page, Mura's consent will only be on the list of cookies once you accept it again. Then it gets a new value "YES%2B20240321-9-57"
But if you refresh the page or browse to a new one, the popup will reappear even though the cookie has been set.  Again, I am not sure this is totally a Masa issue, but it only appeared after the late upgrade. 

First, I have some settings for cookies on my web.config file using the IIS mod rewrite. 

<preConditions>
					<preCondition name="Unsecured cookie">
						<add input="{RESPONSE_SET_COOKIE}" pattern="." />
						<add input="{RESPONSE_SET_COOKIE}" pattern="; secure" negate="true" />
					</preCondition>
				</preConditions>
				<rule name="Remove Server header">
					<match serverVariable="RESPONSE_Server" pattern=".+" />
					<action type="Rewrite" value="" />
				</rule>
				<rule name="Add the STS header in HTTPS responses">
					<match serverVariable="RESPONSE_Strict_Transport_Security" pattern=".*" />
					<conditions>
						<add input="{HTTPS}" pattern="on" />
					</conditions>
					<action type="Rewrite" value="max-age=31536000" />
				</rule>
				<rule name="Use only secure cookies" preCondition="Unsecured cookie">
					<match serverVariable="RESPONSE_SET_COOKIE" pattern=".*" negate="false" />
					<action type="Rewrite" value="{R:0}; secure" />
				</rule>
				<rule name="AddSameSiteCookieFlag">
					<match serverVariable="RESPONSE_Set-Cookie" pattern="^(.*)(CFID|CFTOKEN|__Secure-JSESSIONID|MXP_TRACKINGID|mobileformat)(=.*)$" />
					<action type="Rewrite" value="{R:0}; SameSite=lax" />
				</rule>
				<rule name="AddSecureCookieFlag">
					<match serverVariable="RESPONSE_Set-Cookie" pattern="^(.*)(CFID|CFTOKEN|__Secure-JSESSIONID|MXP_TRACKINGID|mobileformat)(=.*)$" />
					<action type="Rewrite" value="{R:0}; secure" />
				</rule>

This is a live site, so keeping this on at the moment is a bad idea as it causes issues with people browsing the site.  

**Expected behavior**
Usually, after selecting to accept the cookie consent, the popup should only be shown again if you have cleared your cookies.

**Desktop (please complete the following information):**
Every browser both Windows and Mac operating systems

**Environment (please complete the following information):**
 - OS: [ Linux]
 - CFML Engine [Lucee 5.3, Lucee 5.4, Lucee 6.]
 - Database [MSSQL]

**Comment by guustnieuwenhuis** [comment]
Are you talking about www.cookieconsent.com ?

**Comment by jeroenweareorangeeu** [comment]
Looked into this, but I don't know what you mean? 'Built in Cookie Consent' doesnt ring a bell at the moment. Can you provide more details of what you have created / setup in Masa? One tip: if you make changes to your the contentRenderer.cfc in your Masa Theme, then you might have to reload the application for the changes to have effect.

**Comment by Tigerlady** [comment]
So in Masa / site / default directory there is a contentrender.cfc file on about line 260 or so is this code.
		//cookie_consent/modules/cookie_consent_cta/index.cfm
		this.cookieConsentEnabled=true;
		this.cookieConsentType='drawer';
		this.cookieConsentButtonClass="btn btn-primary";
		this.cookieConsentWrapperClass="";
		this.cookieConsentWidth="sm"; //sm,md,lg,full

This is by default set to false; if you turn it to true, you need a component called Cookie Consent. I set this up on all our Masa sites at the very beginning, along with a Privacy Policy for the site. The cookie consent is located in this directory: core/modules/v1/cookie_consent. I turned this on on one of my latest sites, but now it pops up on every page of the site.

**Comment by jeroenweareorangeeu** [comment]
Hello,

Thank you for your further details. Once again I tried to reproduce this.

So in the contentRenderer.cfc of my Theme, I enabled the  Cookie Consent:

`this.cookieConsentEnabled=true`

Then in the Masa Admin I created a Cookie Consent Component:

![image](https://github.com/MasaCMS/MasaCMS/assets/83004496/46891f6c-27bc-48cb-8087-b6054d8bfc68)

Important: the name must be 'Cookie Consent'

Then on my website the Cookie Consent is shown:

![image](https://github.com/MasaCMS/MasaCMS/assets/83004496/e38e567e-a788-47e6-99bf-a844fc87c244)

After clicking 'Got it!' then the popup is not shown anymore. The Cookie Consent functionality seems to work just fine on Masa.

This showing / not showing of the Cookie Consent popup is controled by the following cookie that is set in the browser:

![image](https://github.com/MasaCMS/MasaCMS/assets/83004496/6f4dcfa1-7ddf-4410-b6cd-0ddf91fdfcad)

I would suggest looking into this further, if the cookies are set properly and persisted. In my experience these things might not work properly of webserver settings. I would also suggest reviewing the cookie settings in the settings.ini.cfm. 

Hope this helps you further on your path for solving this issue.
